src/Selenium2Driver.php

```diff
@@ -1120,4 +1132,52 @@ private function trigger($xpath, $event, $options = '{}')
         $script = 'Syn.trigger("' . $event . '", ' . $options . ', {{ELEMENT}})';
         $this->withSyn()->executeJsOnXpath($xpath, $script);
     }
+
+    /**
+     * Uploads a file to the Selenium instance.
+     *
+     * Note that uploading files is not part of the official WebDriver
+     * specification, but it is supported by Selenium.
+     * @see https://github.com/SeleniumHQ/selenium/blob/master/py/selenium/webdriver/remote/webelement.py#L533
+     *
+     * @param string $path     The path to the file to upload.
+     *
+     * @return string          The remote path.
+     *
+     * @throws DriverException When the file is not found.
+     * @throws InvalidRequest  When the driver does not support file uploads.
+     */
+    public function uploadFile($path) {
+        if (!is_file($path)) {
+          throw new DriverException('Could not upload file, the file: ' . $path . '. was not found.');
+        }
+
+        // Selenium only accepts uploads that are compressed as a Zip archive.
+        $temp_filename = tempnam('', 'WebDriverZip');
+
+        $archive = new \ZipArchive();
+        $archive->open($temp_filename, \ZipArchive::CREATE);
+        $archive->addFile($path, basename($path));
+        $archive->close();
+
+        try {
+          $remote_path = $this->getWebDriverSession()->file(array('file' => base64_encode(file_get_contents($temp_filename))));
+        }
+        catch (InvalidRequest $e) {
```

Comment:
Is `InvalidRequest` the only possible error from file upload operation?

Maybe we should catch and rethrow all exceptions?